libchisel/src/lib.rs

```diff
@@ -212,7 +225,12 @@ mod tests {
         let validator = SampleModule {};
         assert_eq!(validator.id(), "Sample");
 
-        let as_trait = validator.as_abstract();
+        let opaque: &dyn ChiselModule<ObjectReference = &dyn ModuleValidator> =
+            &validator as &dyn ChiselModule<ObjectReference = &dyn ModuleValidator>;
```

Comment:
These 2 lines look really nasty.

Reply:
They can be removed in normal code. This is just super explicit so that I know what type is being produced from the expression.